Sources/PropertyEditor.swift

```diff
@@ -0,0 +1,680 @@
+import Result
+
+/// A mutable, observable property that has an optionally failable action
```

Comment:
Unless I'm missing the point, this might instead read:

> An object that manages edits to an underlying MutableProperty using a setter action that can (optionally) validate proposed values.